### PR TITLE
Benchmarks: Code Revision - Revise result process interface and add result checking

### PR DESCRIPTION
**Pull request by guoshzhao**
Major Revisions
1. Modify result process interface to add more check and return True/False.
2. For MicroBenchmark, there have two function, _process_numeric_result() is to process result from python scripts. _process_raw_result is to process raw string result from system call benchmarks.
3. For DockerBenchmark, only _process_raw_result is needed to process raw string result.
4. For ModelBenchmark, only __process_model_result is needed to process result from python scripts.

### CODE REVIEW

**Comment by cp5555** [comment]
Add description following:

**Major Revisions**
- [Revision 1]
- [Revision 2]

**Minor Revisions**
- [Revision 1]
- [Revision 2]

**Review by cp5555 — APPROVED** [review]

File: superbench/benchmarks/micro_benchmarks/micro_base.py [hunk 1]
```diff
@@ -40,17 +41,46 @@ def _preprocess(self):
...
+
+    def _process_raw_result(self, raw_output):
         """Function to process raw results and save the summarized results.
 
         Args:
-            output (str): raw output string of the micro-benchmark.
+            raw_output (str): raw output string of the micro-benchmark.
+
+        Return:
+            True if the raw output string is valid and result can be extracted.
```
Comment (abuccts) [inline superbench/benchmarks/micro_benchmarks/micro_base.py:80]: will this function save results? how does it know where to save?

**Review by abuccts — APPROVED** [review]

File: superbench/benchmarks/micro_benchmarks/micro_base.py (hunk 1 above)
Comment (guoshzhao) [reply superbench/benchmarks/micro_benchmarks/micro_base.py:80]: This function also need to call the following function to save result:
        self._result.add_raw_data()
        self._result.add_result()
       
Benchmark owner need to know this. I will add more comments.

**Comment by guoshzhao** [comment]
> Add description following:
> 
> **Major Revisions**
> 
> * [Revision 1]
> * [Revision 2]
> 
> **Minor Revisions**
> 
> * [Revision 1]
> * [Revision 2]

Done.

**Pull request merged**